analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -0,0 +1,316 @@
+                    vcs = projectPackage.vcs,
+                    vcsProcessed = processProjectVcs(workingDir, projectPackage.vcs, projectPackage.homepageUrl),
+                    homepageUrl = projectPackage.homepageUrl,
+                    scopes = sortedSetOf(dependenciesScope, devDependenciesScope)
+                ),
+                packages = packages.map { it.value.toCuratedPackage() }.toSortedSet()
+            )
+        }
+    }
+
+    private fun extractDependencyTree(
+        rootNode: JsonNode,
+        workingDir: File,
+        pkg: JsonNode,
+        scopeName: String
+    ): SortedSet<PackageReference> {
+        val result = mutableSetOf<PackageReference>()
+        pkg[ scopeName ]?.forEach {
+            val childRef = it.textValueOrEmpty()
+            rootNode.iterator().forEach { child ->
+                if (child["reference"].textValueOrEmpty() == childRef) {
+                    log.debug { "Found child. '$childRef'" }
+
+                    val packageReference = PackageReference(
+                        id = extractPackageId(child, workingDir),
+                        dependencies = extractDependencyTree(rootNode, workingDir, child, SCOPE_NAME_DEPENDENCIES)
+                    )
+                    result.add(packageReference)
+
+                    val packageDevReference = PackageReference(
+                        id = extractPackageId(child, workingDir),
+                        dependencies = extractDependencyTree(rootNode, workingDir, child, SCOPE_NAME_DEV_DEPENDENCIES)
+                    )
+                    result.add(packageDevReference)
+                }
+            }
+        }
+        return result.toSortedSet()
+    }
+
+    /**
+     * Runs through each package and extracts list of its dependencies (also transitive ones)
+     */
+    private fun extractDependencies(
+        rootNode: JsonNode,
+        scopeName: String,
+        workingDir: File
+    ): SortedSet<PackageReference> {
+        val stack = Stack<JsonNode>()
+        val dependencies = mutableSetOf<PackageReference>()
+
+        stack.addAll(rootNode)
+        while (!stack.empty()) {
+            val pkg = stack.pop()
+            extractDependencyTree(rootNode, workingDir, pkg, scopeName).forEach {
+                dependencies.add(it)
+            }
+        }
+        return dependencies.toSortedSet()
+    }
+
+    /**
+     * Extracts plain list of all packages
+     */
+    private fun extractPackages(node: List<JsonNode>, workingDir: File): Map<String, Package> {
+        val result = mutableMapOf<String, Package>()
+        val stack = Stack<JsonNode>()
+        stack.addAll(node)
+        while (!stack.empty()) {
+            val currentNode = stack.pop()
+            val pkg = extractPackage(currentNode, workingDir)
+            result["${pkg.id.name}:${pkg.id.version}"] = pkg
+        }
+        return result
+    }
+
+    /**
+     * Creates corresponding `Package` object from given `JsonNode` object
```

Comment:
Nit: Imperative mood. Dot at end of sentence.

Reply:
Also here: If you add function documentation, please also document all function parameters (in KDoc style, not JavaDoc style, i.e. use `[node]`, not `@param node`).